components/core/src/clp_s/CommandLineArguments.cpp

```diff
@@ -382,15 +417,38 @@ CommandLineArguments::parse_arguments(int argc, char const** argv) {
 
                 po::options_description visible_options;
                 visible_options.add(general_options);
-                visible_options.add(input_options);
                 visible_options.add(decompression_options);
                 visible_options.add(output_metadata_options);
                 std::cerr << visible_options << std::endl;
                 return ParsingResult::InfoCommand;
             }
 
-            if (m_archives_dir.empty()) {
-                throw std::invalid_argument("No archives directory specified");
+            if (archive_path.empty()) {
+                throw std::invalid_argument("No archive path specified");
+            }
+
+            if (false == archive_id.empty()) {
```

Comment:
It seems that without archive id, the path can be either a url for a filesystem path, but with archive id, only filesystem paths are allowed?

Reply:
Yes -- I considered taking out the archive id option entirely, but decided to leave it in to maintain our existing behaviour for archives on a filesystem. Not really any point in supporting the archive-id option for network addresses since we just figure out the archive id based on the url.